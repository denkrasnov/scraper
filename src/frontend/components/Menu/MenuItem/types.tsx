export interface MenuItemProps {
  src: string;
  text: string;
}
