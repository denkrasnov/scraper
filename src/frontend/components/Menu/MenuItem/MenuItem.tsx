import React, { FC } from "react";

import Box from "~app/atoms/Box";
import Image from "~app/atoms/Image";
import { H3 } from "~app/atoms/Headings";
import { MenuItemProps } from "./types";

const MenuItem: FC<MenuItemProps> = (props) => {
  const { src, text } = props;

  return (
    <Box
      cursor="pointer"
      flexDirection="column"
      marginRight="60px"
      width="200px"
    >
      <Image alt="menu-image" height="134px" src={src} />
      <Box justifyContent="center" width="100%">
        <H3>{text}</H3>
      </Box>
    </Box>
  );
};

export default MenuItem;
