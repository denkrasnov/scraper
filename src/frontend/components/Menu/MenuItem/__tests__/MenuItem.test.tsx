import React from "react";
import { shallow } from "enzyme";

import MenuItem from "..";

describe("MenuItem", () => {
  it("should render", () => {
    const component = shallow(<MenuItem src="/__SRC__" text="__TEXT__" />);

    expect(component).toMatchSnapshot();
  });
});
