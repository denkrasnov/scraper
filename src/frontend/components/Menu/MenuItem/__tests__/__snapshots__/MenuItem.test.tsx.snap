// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MenuItem should render 1`] = `
<Box
  cursor="pointer"
  flexDirection="column"
  marginRight="60px"
  width="200px"
>
  <Image
    alt="menu-image"
    height="134px"
    src="/__SRC__"
  />
  <Box
    justifyContent="center"
    width="100%"
  >
    <H3>
      __TEXT__
    </H3>
  </Box>
</Box>
`;
